xbmc/cores/VideoPlayer/Edl.cpp

```diff
@@ -184,376 +181,373 @@ bool CEdl::ReadEdl(const std::string& strMovie, const float fFramesPerSecond)
         }
         else
         {
-          bError = true;
+          error = true;
           continue;
         }
       }
-      else if (strFields[i][0] == '#') // #12345 format for frame number
+      // #12345 format for frame number
+      else if (fields[i][0] == '#')
       {
-        if (fFramesPerSecond > 0.0f)
+        if (fps > 0.0f)
         {
-          editStartEnd[i] = static_cast<int64_t>(std::atol(strFields[i].substr(1).c_str()) /
-                                                 fFramesPerSecond * 1000); // frame number to ms
+          editStartEnd[i] = static_cast<int64_t>(std::atol(fields[i].substr(1).c_str()) / fps *
+                                                 1000); // frame number to ms
         }
         else
         {
-          CLog::Log(LOGERROR,
-                    "Edl::ReadEdl - Frame number not supported in EDL files when frame rate is "
-                    "unavailable (ts) - supplied frame number: {}",
-                    strFields[i].substr(1));
+          CLog::LogF(LOGERROR,
+                     "Frame number not supported in EDL files when frame rate is "
+                     "unavailable (ts) - supplied frame number: {}",
+                     fields[i].substr(1));
           return false;
         }
       }
-      else // Plain old seconds in float format, e.g. 123.45
+      // Plain old seconds in float format, e.g. 123.45
+      else
       {
         editStartEnd[i] =
-            static_cast<int64_t>(std::atof(strFields[i].c_str()) * 1000); // seconds to ms
+            static_cast<int64_t>(std::atof(fields[i].c_str()) * 1000); // seconds to ms
       }
     }
 
-    if (bError) // If there was an error in the for loop, ignore and continue with the next line
+    // If there was an error in the for loop, ignore and continue with the next line
+    if (error)
       continue;
 
     Edit edit;
     edit.start = editStartEnd[0];
     edit.end = editStartEnd[1];
 
-    switch (iAction)
+    switch (action)
```

Comment:
This switch/case is pretty ugly, Is there the possibility (in the future) to map the action int to an enum class? This would make it more clear.

Reply:
Yeah I also don't like this, I think I'll handle this in this PR